src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -51,7 +54,17 @@ internal sealed class MethodBuilderImpl : MethodBuilder
 
         protected override bool InitLocalsCore { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
         protected override GenericTypeParameterBuilder[] DefineGenericParametersCore(params string[] names) => throw new NotImplementedException();
-        protected override ParameterBuilder DefineParameterCore(int position, ParameterAttributes attributes, string? strParamName) => throw new NotImplementedException();
+        protected override ParameterBuilder DefineParameterCore(int position, ParameterAttributes attributes, string? strParamName)
+        {
+            if (position > 0 && (_parameterTypes == null || position > _parameterTypes.Length))
+                throw new ArgumentOutOfRangeException(SR.ArgumentOutOfRange_ParamSequence);
+
+            Debug.Assert(_parameters != null);
```

Comment:
Is this guaranteed? I think you can end up with `_parameters = null` when position argument is 0

Reply:
Right